### PR TITLE
Scouting ntuplizers

### PR DESCRIPTION
**Pull request by gpetruc**
* ntuplizers for the W3pi, multijet and di-muon level1-scouting studies
 * dumper to create raw data files with the puppi objects and puppi jets (the latter hasn't really been tested yet)

### CODE REVIEW

File: NtupleProducer/python/scouting/runW3PiNtuples_cfg.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+process.p.associate(process.extraPFStuff)
+
+def goSignal():
+    ## Add to the configuration:
+    ##   * the generator-level W boson and pions
+    ##   * for each Puppi candidate, information on whether it's matched to a gen pion
+    ##   * the table of PF candidates, so that one can 
+    process.genW = cms.EDFilter("GenParticleSelector",
+            src = cms.InputTag("genParticles"),
+            cut = cms.string("abs(pdgId) = 24 && numberOfDaughters = 3 && abs(daughter(0).pdgId) = 211 && abs(daughter(1).pdgId) = 211"),
```
Comment (francescobrivio) [inline NtupleProducer/python/scouting/runW3PiNtuples_cfg.py]: Shouldn't this cut consider the 3<sup>rd</sup> daughter as well?
```suggestion
            cut = cms.string("abs(pdgId) = 24 && numberOfDaughters = 3 && abs(daughter(0).pdgId) = 211 && abs(daughter(1).pdgId) = 211 && abs(daughter(2).pdgId) = 211"),
```

**Comment by francescobrivio** [comment]
I also tried running `cmsRun runW3PiNtuples_cfg.py`, but I get the following error:
```
----- Begin Fatal Exception 28-Oct-2023 23:26:14 CEST-----------------------
An exception of category 'ProductNotFound' occurred while
   [0] Processing  Event run: 1 lumi: 1000 event: 3 stream: 0
   [1] Running path 'p'
   [2] Prefetching for module SimpleCandidateFlatTableProducer/'puppiTable'
   [3] Prefetching for module DeregionizerProducer/'l1tLayer2Deregionizer'
   [4] Prefetching for module L1TPFCandMultiMerger/'l1tLayer1'
   [5] Prefetching for module L1TCorrelatorLayer1Producer/'l1tLayer1Barrel'
   [6] Calling method for module VertexProducer/'l1tVertexFinderEmulator'
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for type: edm::RefVector<std::vector<TTTrack<edm::Ref<edm::DetSetVector<Phase2TrackerDigi>,Phase2TrackerDigi,edm::refhelper::FindForDetSetVector<Phase2TrackerDigi> > > >,TTTrack<edm::Ref<edm::DetSetVector<Phase2TrackerDigi>,Phase2TrackerDigi,edm::refhelper::FindForDetSetVector<Phase2TrackerDigi> > >,edm::refhelper::FindUsingAdvance<std::vector<TTTrack<edm::Ref<edm::DetSetVector<Phase2TrackerDigi>,Phase2TrackerDigi,edm::refhelper::FindForDetSetVector<Phase2TrackerDigi> > > >,TTTrack<edm::Ref<edm::DetSetVector<Phase2TrackerDigi>,Phase2TrackerDigi,edm::refhelper::FindForDetSetVector<Phase2TrackerDigi> > > > >
Looking for module label: l1tTrackSelectionProducer
Looking for productInstanceName: Level1TTTracksSelectedEmulation

   Additional Info:
      [a] If you wish to continue processing events after a ProductNotFound exception,
add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.

----- End Fatal Exception -------------------------------------------------
```
And I get the same error from `runJetStudyNTuple.py`.
No idea how to debug this 😄  (probably a producer for `Level1TTTracksSelectedEmulation` is missing)

On the contrary, `runMuNTuple_cfg.py` works and produces a filled ntuple.

File: NtupleProducer/python/scouting/runW3PiNtuples_cfg.py (hunk 1 above)
Comment (gpetruc) [reply NtupleProducer/python/scouting/runW3PiNtuples_cfg.py]: well, in these MC events the W's either decay to 3 pions or have some "internal" decay like W -> W + gamma that pythia uses to denote radiation, so checking 2 was already sufficient, but indeed checking the third doesn't hurt and makes it more readable

**Comment by gpetruc** [comment]
Now all ntuplizers are updates and run fine on the current L1PF_12_5_X branch (tag: l1ct-125x-v1.21).

**Pull request merged**